src/components/list/ListForm.vue

```diff
@@ -0,0 +1,264 @@
+            categories
+          </h1>
+          <div
+            class="filter"
+            v-for="(category, index) in list.filters.categories"
+            :key="index"
+          >
+            <InputCustom
+              v-model="category.name"
+            />
+            <div
+              class="delete-filter-button"
+              @click="deleteFilter('categories', index)"
+            ></div>
+          </div>
+          <div
+            class="add-filter-button"
+            @click="addFilter('categories')"
+          ></div>
+        </div>
+      </div>
+      <ErrorMessage
+        :message="errorMessage"
+        v-if="errorMessage"
+      />
+      <ButtonText
+        text="delete list"
+        style-type="bordered"
+        v-if="edittingListObj"
+        @click="deleteList(list.id)"
+      />
+    </template>
+    <template v-slot:buttons>
+      <ButtonText
+        text="save"
+        style-type="bordered"
+        @click="submitList"
+      />
+      <ButtonText
+        text="cansel"
+        style-type="bordered"
+        @click="closeListForm"
+      />
+    </template>
+  </ModalBasic>
+</template>
+
+<script>
+import ModalBasic from '@/components/modals/ModalBasic.vue';
+import InputCustom from '@/components/formElements/InputCustom.vue';
+import CheckboxCustom from '@/components/formElements/CheckboxCustom.vue';
+import ButtonText from '@/components/formElements/ButtonText.vue';
+import ErrorMessage from '@/components/textElements/ErrorMessage.vue';
+import models from '@/models/models';
+import { mapActions, mapGetters } from 'vuex';
+
+export default {
+  components: {
+    ModalBasic,
+    InputCustom,
+    CheckboxCustom,
+    ButtonText,
+    ErrorMessage,
+  },
+  data: () => ({
+    list: new models.List(),
+    errorMessage: '',
+  }),
+  computed: {
+    ...mapGetters({
+      edittingListObj: 'edittingListObj',
+    }),
+  },
+  methods: {
+    ...mapActions({
+      _setListForEditting: '_setListForEditting',
+      _addList: '_addList',
+      _saveList: '_saveList',
+      _removeList: '_removeList',
+    }),
+    closeListForm() {
+      this.$modal.hide('listForm');
+    },
+    setListForEditting() {
+      if (this.edittingListObj) {
+        this.list = { ...this.edittingListObj };
+      }
+    },
+    clearData() {
+      this._setListForEditting(null);
+      this.list = new models.List();
+      this.errorMessage = '';
+    },
+    deleteFilter(type, index) {
+      this.list.filters[type].splice(index, 1);
+    },
+    addFilter(type) {
+      this.list.filters[type].push({
+        name: null,
+        id: null,
+      });
+    },
+    verifyFilters() {
+      const filtersNames = [];
+      this.list.filters.tags.forEach(tag => filtersNames.push(tag.name));
+      this.list.filters.categories.forEach(category => filtersNames.push(category.name));
+      const noRepeatFiltersNames = new Set(filtersNames);
+
+      return filtersNames.length === noRepeatFiltersNames.size;
```

Comment:
Interesting solution. cool to use Set here!

but what about more greedy algorithm? Like can we make this checking shorter? After first finding repeating?